crates/re_types_builder/src/context.rs

```diff
@@ -0,0 +1,125 @@
+use std::sync::mpsc;
+
+/// Creates a new context.
+///
+/// The [`Context`] can be freely cloned and sent to other threads.
+///
+/// The [`ContextRoot`] should not be sent to other threads.
+pub fn context() -> (ContextRoot, Context) {
+    let (tx, rx) = mpsc::channel();
+    (ContextRoot::new(rx), Context::new(tx))
+}
+
+#[derive(Clone)]
+pub struct Context {
+    errors: mpsc::Sender<anyhow::Error>,
+}
+
+impl Context {
+    fn new(errors: mpsc::Sender<anyhow::Error>) -> Self {
+        Self { errors }
+    }
+
+    pub fn error(&self, error: impl IntoError) {
+        let _ = self.errors.send(error.into_error());
+    }
+}
+
+pub struct ContextRoot {
+    errors: mpsc::Receiver<anyhow::Error>,
+    _not_send: std::marker::PhantomData<*mut ()>,
+}
+
+impl ContextRoot {
+    fn new(errors: mpsc::Receiver<anyhow::Error>) -> Self {
+        Self {
+            errors,
+            _not_send: std::marker::PhantomData,
+        }
+    }
+
+    /// This outputs all errors to stderr and panics if there were any.
+    pub fn panic_on_errors(&self) {
+        let mut errored = false;
+
+        while let Ok(err) = self.errors.try_recv() {
+            errored = true;
+            eprintln!("{err}");
+        }
+
+        #[allow(clippy::manual_assert)] // we don't want the noise of an assert
+        if errored {
+            panic!("Some errors occurred.");
```

Comment:
The errors must be part of the panic, or it won't be visible when run inside a `build.rs`

Reply:
Really? I thought the `build.rs` will print captured stdio when it panics. At least that's the behavior I observed:

```rust
$ cargo rerun
  <... cargo:rerun-if-changed spam>

  --- stderr
  re_types_builder_hash: "f4d9ee9d23151340d6284943c06776ddd1966eaf12089367a4d8c4d000cd4fa4"
  definitions_hash: "22c9802f61edfe625d911ca849cbe06027a7d06e220068efb521a2b1dd27d3d4"
  doc_examples_hash: "79b7d3182fec4b6cc18cb690f93f816af343ec2e62c8741e57dde3fb45ae6bfb"
  python_extensions_hash: "8658b955f2803c5d6b06c4f7e690a18d6adb8483cda72a563e17ab7f7238a80e"
  cpp_extensions_hash: "173be500018fd28848100fc8502214765b6f35a4f08e04e22c41769ed9749d6c"
  new_hash: "b4642838983a397040bd43202b5eb772e9522817fe828ff9706d3e124e2488a4"
  cur_hash: Some("99644a08283d24ba10136dfa1b029874a01c13f1b3973f292c303535dc3621ca")
  $ flatc -I /home/jp/rerun-io/rerun/crates/re_types/definitions -o /tmp/xshell-tmp-dir-1 -b --bfbs-comments --schema ./definitions/rerun/archetypes.fbs
  This is a sample error
  thread 'main' panicked at 'Some errors occurred.', crates/re_types_builder/src/context.rs:52:13
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

`This is a sample error` comes from `ctx.error`.

Reply:
Oh, cool, TIL!